File: server/iih_pub/iih.ci_pub/src/main/java/iih/ci/mr/nu/obstetrics/afterdeliverobsec/d/AfterdeliverobsecAggDO.java
package iih.ci.mr.nu.obstetrics.afterdeliverobsec.d;

import xap.sys.appfw.orm.model.agg.BaseAggDO;
import xap.mw.core.data.BaseDO;
import xap.mw.coreitf.i.IAggDesc;
import xap.mw.coreitf.i.IBaseDO;
import iih.ci.mr.nu.obstetrics.afterdeliverobsec.d.desc.AfterdeliverobsecAggDODesc;



/**
 * 产后观察记录
 */
public class AfterdeliverobsecAggDO extends BaseAggDO {
	private static final long serialVersionUID = 1L;

	@Override
	public AfterDeliveInfoDO getParentDO() {
		return ((AfterDeliveInfoDO) super.getParentDO());
	}

	public void setParentDO(AfterDeliveInfoDO headDO) {
		setParent(headDO);
	}

	public AfterDeliveRecDO[] getAfterDeliveRecDO() {
		IBaseDO[] dos = getChildren(AfterDeliveRecDO.class);
		if(dos==null) return null;
		AfterDeliveRecDO[] result=new AfterDeliveRecDO[dos.length];
		for(int i=0;i<dos.length;i++){
			result[i]=(AfterDeliveRecDO)dos[i];
		}
		return result;
	}
	
	public void setAfterDeliveRecDO(AfterDeliveRecDO[] dos) {
		setChildren(AfterDeliveRecDO.class, dos);
	}

	@Override
	public IAggDesc getAggDesc() {
		AfterdeliverobsecAggDODesc desc = new AfterdeliverobsecAggDODesc();

		return desc;
	}
	
	 public  BaseDO createParentDO() {
         return new AfterDeliveInfoDO();
     }

     public  BaseDO createChildDO(String clzName) {
	     if (clzName.equals("iih.ci.mr.nu.obstetrics.afterdeliverobsec.d.AfterDeliveRecDO")) {
             return new AfterDeliveRecDO();
         }
         return null;
     }
}

File: client/iih_stub/iih.ci_stub/iih.ci.mr/nu/obstetrics/afterdeliverobsec/d/AfterdeliverobsecAggDO.cs
using System;
using xap.mw.core.data;
using xap.mw.core.meta;

namespace iih.ci.mr.nu.obstetrics.afterdeliverobsec.d
{
	/// <summary>
	/// 产后观察记录DO
	/// </summary>
	public class AfterdeliverobsecAggDO : BaseAggDO
	{	
	    private static string AFTERDELIVEINFODO_AFTERDELIVERECDO="iih.ci.mr.nu.obstetrics.afterdeliverobsec.d.AfterDeliveRecDO";

        public AfterdeliverobsecAggDO() {
            this.setParentDO(new AfterDeliveInfoDO());
        }

  	    public AfterDeliveInfoDO getParentDO() {
		    return ((AfterDeliveInfoDO) base.ParentDO);
	    }      
	    
	    public void setParentDO(AfterDeliveInfoDO headDO) {
		    base.ParentDO=headDO;
	    }
	    
	    public AfterDeliveRecDO[] getAfterDeliveRecDO() {		    
		    
		    BaseDO[] dos=getChildren(Type.GetType(AFTERDELIVEINFODO_AFTERDELIVERECDO));
		    if (dos == null || dos.Length==0){return null;}
            AfterDeliveRecDO[] r = new AfterDeliveRecDO[dos.Length];
            for (int i = 0; i < dos.Length;i++ ){
                r[i] = dos[i] as AfterDeliveRecDO;
            }
	        return r;
		    
	    }
	    
	    public void setAfterDeliveRecDO(AfterDeliveRecDO[] dos) {
            if (dos == null || dos.Length == 0) return;
            setChildren(dos[0].GetType(), dos);
	    }


	    public override BaseDO createParentDO()
	    {
	        return new AfterDeliveInfoDO();
	    }

        public override BaseDO createChildDO(string clzName)
	    {
            if (AFTERDELIVEINFODO_AFTERDELIVERECDO == clzName)
	        {
                return new AfterDeliveRecDO();
	        }
	        return null;
	    }
	    
	    /// <summary>
        /// 获取子类全路径名数组
        /// </summary>
        /// <returns></returns>
        public override string[] getChildClassNames(){
            return  new string[]{"iih.ci.mr.nu.obstetrics.afterdeliverobsec.d.AfterDeliveRecDO"};
        }
	    public new static AggMeta GetMeta()
	    {
	        AggMeta meta = new AggMeta();
	        meta.OneToOneDOs = new string[]{};
	        return meta;
	    }
	    
	}
}
